integration-tests/features/steps/common.py

```diff
@@ -678,3 +685,34 @@ def check_analyzed_dependency(context, package, version):
     else:
         raise Exception('Package {package} with version {version} not found'.
                         format(package=package, version=version))
+
+
+@when('I generate authorization token from the private key {private_key}')
+def generate_authorization_token(context, private_key):
+    expiry = datetime.datetime.utcnow() + datetime.timedelta(days=90)
+    userid = "testuser"
+
+    path_to_private_key = 'data/{private_key}'.format(private_key=private_key)
+    # initial value
+    context.token = None
+
+    with open(path_to_private_key) as fin:
+        private_key = fin.read()
+
+        try:
+            payload = {
+                'exp': expiry,
+                'iat': datetime.datetime.utcnow(),
+                'sub': userid
+            }
+            token = jwt.encode(payload, key=private_key, algorithm='RS256')
+            decoded = token.decode('utf-8')
+            # print(decoded)
+            context.token = decoded
+        except Exception as e:
+            print("Error occured: " + e)
```

Comment:
Thanks for review. Actually I would not like to propagate failures, I'm updating the code.